packages/driver/src/cy/commands/querying.js

```diff
@@ -277,7 +277,17 @@ module.exports = (Commands, Cypress, cy) => {
         let $el
 
         try {
-          $el = cy.$$(selector, options.withinSubject)
+          // only support shadow traversal if we're not searching
+          // within a subject and have been explicitly told to ignore
+          // boundaries.
+          if (!options.ignoreShadowBoundaries) {
+            $el = cy.$$(selector, options.withinSubject)
+          } else {
+            const elementsWithShadow = $dom.findAllShadowRoots(options.withinSubject || cy.state('document'))
```

Comment:
@chrisbreiding would this maybe work? find all the roots within the subject if it is set, otherwise use the document. then select within those results (which includes the subject its self)

Reply:
Yeah, I think that should work. Thanks!